server/src/plugins/groups-decorator.ts

```diff
@@ -0,0 +1,46 @@
+import fp from 'fastify-plugin';
+import { FastifyReply, FastifyInstance } from 'fastify';
+import { IncomingMessage, ServerResponse, Server } from 'http';
+import { UserRole, auth0USer } from '../types';
+
+export class jwtIncomingMessage extends IncomingMessage {
+  user?: {
+    'https://poap.xyz/roles': [ string ];
+    iss: string;
+    sub: string;
+    aud: [ string, ];
+    iat: Date;
+    exp: Date;
+    azp: string;
+    scope: string;
+  }
+}
+
+export function getUserRoles(user: auth0USer | any){
+  return user['https://poap.xyz/roles']
+}
+
+declare module 'fastify' {
+  export interface FastifyInstance<HttpServer = Server, HttpRequest = IncomingMessage, HttpResponse = ServerResponse> {
+    isAdmin: any;
+  }
+}
+
+export default fp((fastify: FastifyInstance<Server, IncomingMessage, ServerResponse>, opts, next) => {
+    fastify.decorate('isAdmin', 
+      async (request: jwtIncomingMessage, reply: FastifyReply<ServerResponse>) => {
+        try {
+          if(!request.user) {
+            reply.send('Invalid user');
+          }
+          if (request.user && getUserRoles(request.user).indexOf(UserRole.administrator) == -1) {
+            reply.send('User is not allowed');
+          }
```

Comment:
Optionally you can use `includes` method and move that condition into a constant.
Also we already check if there's no user on the request object and send a reply so I guess there's no need to check if exists since we already exit the function with the `reply.send()` from line 33.

Code example:
```typescript
const isUserAdmin = getUserRoles(request.user).includes(UserRole.administrator);

if (!isUserAdmin) {
  reply.send('User is not allowed');
}
```